app/src/backend/ActivityService.ts

```diff
@@ -21,9 +21,225 @@ import type { Transaction } from "sequelize";
 
 type GasValueInput = Omit<GasValueCreationAttributes, "id"> & {
   emissionsFactor?: Omit<EmissionsFactorAttributes, "id">;
+  id?: string;
+};
+
+export type UpdateGasValueInput = GasValueCreationAttributes & {
+  emissionsFactor?: EmissionsFactorAttributes;
 };
 
 export default class ActivityService {
+  private static async updateDataSource({
+    activityValue,
+    dataSourceParams,
+    transaction,
+  }: {
+    activityValue: ActivityValue;
+    dataSourceParams: Omit<DataSourceAttributes, "datasourceId"> | undefined;
+    transaction: Transaction;
+  }): Promise<string> {
+    const dataSource = await db.models.DataSource.findOne({
+      where: { datasourceId: activityValue.datasourceId },
+      transaction,
+    });
+
+    if (!dataSource) {
+      throw new createHttpError.NotFound(
+        "Data source for ActivityValue not found",
+      );
+    }
+
+    // update the data source
+    await dataSource.update(
+      {
+        ...dataSourceParams,
+      },
+      { transaction },
+    );
+    return dataSource.datasourceId;
+  }
+
+  private static async updateInventoryValue({
+    activityValue,
+    inventoryValueParams,
+    transaction,
+  }: {
+    activityValue: ActivityValue;
+    inventoryValueParams?: Omit<InventoryValueAttributes, "id"> | undefined;
+    transaction: Transaction;
+  }) {
+    let inventoryValueId = activityValue.inventoryValueId;
+
+    let inventoryValue = await db.models.InventoryValue.findByPk(
+      inventoryValueId,
+      { transaction },
+    );
+
+    if (!inventoryValue) {
+      throw new createHttpError.NotFound(
+        `Inventory value with ID ${inventoryValueId} not found`,
+      );
+    }
+
+    // always make sure an input methodology exists
+    if (
+      !(
+        inventoryValue.inputMethodology ||
+        inventoryValueParams?.inputMethodology
+      )
+    ) {
+      throw new createHttpError.NotFound(
+        `Inventory value ${inventoryValue.id} is missing an input methodology`,
+      );
+    }
+
+    // if inventoryValueParams exist update the inventoryValues
+    if (inventoryValueParams) {
+      const { sectorId, subSectorId, subCategoryId } =
+        await GPCService.getIDsFromReferenceNumber(
+          inventoryValueParams.gpcReferenceNumber!,
+        );
```

Comment:
Not sure if we need to support moving it to a different subsector/ sector etc. after creating it.
Doesn't seem like a requirement from the UI side, so maybe we can avoid the additional query...